aws_lambda_builders/workflows/nodejs_npm_esbuild/workflow.py

```diff
@@ -53,150 +50,75 @@ def __init__(self, source_dir, artifacts_dir, scratch_dir, manifest_path, runtim
-            EsbuildCheckVersionAction(scratch_dir, subprocess_esbuild),
-            EsbuildBundleAction(
-                scratch_dir,
-                artifacts_dir,
-                bundler_config,
-                osutils,
-                subprocess_esbuild,
-                self.manifest_path,
-                skip_deps=True,
-            ),
-        ]
-        esbuild_with_deps = EsbuildBundleAction(
-            scratch_dir, artifacts_dir, bundler_config, osutils, subprocess_esbuild, self.manifest_path
-        )
+        if not self.download_dependencies and not self.dependencies_dir:
+            # Invalid workflow, can't have no dependency dir and no installation
+            raise EsbuildExecutionError(
+                message="Lambda Builders encountered an invalid workflow. A workflow can't "
+                "include a dependencies directory without installing dependencies."
+            )
 
-        install_action = NodejsNpmWorkflow.get_install_action(
-            source_dir,
-            scratch_dir,
-            subprocess_npm,
-            osutils,
-            self.options,
+        self.actions.append(
+            CopySourceAction(source_dir=self.source_dir, dest_dir=self.scratch_dir, excludes=self.EXCLUDED_FILES)
         )
 
-        if self.download_dependencies and not self.dependencies_dir:
-            return actions + [install_action, esbuild_with_deps]
-
-        return self._accelerate_workflow_actions(
-            source_dir, scratch_dir, actions, install_action, esbuild_with_deps, esbuild_no_deps
+        bundle_action = EsbuildBundleAction(
+            working_directory=self.scratch_dir,
+            output_directory=self.artifacts_dir,
+            bundler_config=bundler_config,
+            osutils=self.osutils,
+            subprocess_esbuild=self.subprocess_esbuild,
+            manifest=self.manifest_path,
+            skip_deps=not self.combine_dependencies,
         )
 
-    def _accelerate_workflow_actions(
-        self, source_dir, scratch_dir, actions, install_action, esbuild_with_deps, esbuild_no_deps
-    ):
-        """
-        Generate a list of Nodejs build actions for incremental build and auto dependency layer
-
-        :type source_dir: str
-        :param source_dir: an existing (readable) directory containing source files
-
-        :type scratch_dir: str
-        :param scratch_dir: an existing (writable) directory for temporary files
-
-        :type actions: List[BaseAction]
-        :param actions: List of existing actions
-
-        :type install_action: BaseAction
-        :param install_action: Installation action for npm
-
-        :type esbuild_with_deps: BaseAction
-        :param esbuild_with_deps: Standard esbuild action bundling source with deps
+        if self.download_dependencies:
+            self.actions += [
+                NodejsNpmWorkflow.get_install_action(
+                    source_dir=source_dir,
+                    artifacts_dir=self.scratch_dir,
```

Comment:
it would be better if this argument was named install_dir, as it actually controls where dependencies are installed. Will rename in another PR, when I introduce building-in-source to esbuild